File: front/front-vue/src/api/PostService.js
import Api from "./Api";

class PostService {
  async getPost(post_no) {
    return Api.get(`/Post/search/${post_no}`)
      .then(Response => {
        return Response;
      })
      .catch(exp => {
        console.log("Error getPost : " + exp);
      });
  }
  async getPostDetail(post_no) {
    return Api.get(`/Post/searchDetail/${post_no}`)
      .then(Response => {
        return Response;
      })
      .catch(exp => {
        console.log("Error getPostDetail : " + exp);
      });
  }
  async insertPost(post) {
    return Api.post("/Post/insert", post)
      .then(() => {
        // console.log(Response);
      })
      .catch(exp => {
        console.log("Error insertPost : " + exp);
      });
  }
  async updatePost(post) {
    return Api.put("/Post/update", post)
      .then(() => {
        // console.log(Response);
      })
      .catch(exp => {
        console.log("Error updatePost : " + exp);
      });
  }
  async deletePost(post_no) {
    return Api.delete(`/Post/delete/${post_no}`)
      .then(() => {
        // console.log(Response);
      })
      .catch(exp => {
        console.log("Error deletePost : " + exp);
      });
  }
}
export default new PostService();

File: back/back-spring/src/main/java/com/ssafy/backspring/model/service/PostService.java
package com.ssafy.backspring.model.service;

import java.util.List;

import com.ssafy.backspring.model.dto.Post;

public interface PostService {
	public void insert(Post post);
	public Post search(int post_no);
	public List<Post> searchAll();
	public void update(Post post);
	public void delete(int post_no);
	public Post searchPostDetail(int post_no);
	public void addPostViews(int post_no);
	public void addPostRecommendation(int post_no);
}
